File: src/first_package/service/add_number_server.cpp
#include "ros/ros.h"
#include "first_package/add_number.h"

bool add(first_package::add_number::Request  &req,
         first_package::add_number::Response &res)
{
  res.total = (req.x * req.y);
  ROS_INFO("request: x=%ld, y=%ld", (long int)req.x, (long int)req.y);
  ROS_INFO("sending back response: [%ld]", (long int)res.total);
  return true;
}

int main(int argc, char **argv)
{
    //init the node name and nodehandle 
    ros::init(argc, argv, "add_two_number_server");
    ros::NodeHandle n;
    
    //defind the server <object> = advertise server name and add function
    //jump to add functiuon
    ros::ServiceServer service = n.advertiseService("add_two_number", add);
    ROS_INFO("Ready to add two ints.");
    ros::spin();
    
    return 0;
}

File: src/first_package/service/add_number_server.py
#!/usr/bin/env python
from first_package.srv import add_number
from first_package.srv import add_numberRequest
from first_package.srv import add_numberResponse

import rospy

def handle_add_two_number(req):
    print("Returning [%s + %s = %s]"%(req.x, req.y, (req.x+req.y)))
    #return the sum of req.x, req.y to Response
    return add_numberResponse(req.x+req.y)

#create the server
def add_two_number_server():
    #define a node name
    rospy.init_node("add_two_number_server")
    #throw information to the service (request)
    #add_two_number service name, service type add_number, callbake function handle_add_two_number
    server = rospy.Service("add_two_number", add_number, handle_add_two_number)
    print("Ready to add two number")
    rospy.spin()

if __name__ == "__main__":
    add_two_number_server()
